packages/extension-starter/src/setup-app.ts

```diff
@@ -79,14 +81,19 @@ class Sample extends ApplicationExtension<Config> {
      *
      * NOTE: If you instead want to modify a list of all the routes, refer to the `beforeRouteMatch` below.
      */
-    getRoutes(params: GetRoutesParams): RouteProps[] {
-        return [
+    getRoutesAsync(params: GetRoutesParams): Promise<AsyncRouteProps[]> {
+        return Promise.resolve([
             {
                 exact: true,
                 path: this.getConfig().path,
-                component: SamplePage
+                component: SamplePage,
+                componentProps: {title: 'A custom title from the extension'}
             }
-        ]
+        ])
+    }
+
+    getComponentMap(): ComponentMap {
+        return {SamplePage}
```

Comment:
here is an example of how regular extension's implementing `getRoutesAsync` would look like while passing props